nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2078,6 +2080,66 @@ async fn instance_create(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Path parameters for Instance requests. Name is temporary.
+#[derive(Deserialize, JsonSchema)]
+struct InstanceLookupPathParam {
+    instance: ResourceIdentifier,
+}
+
+#[derive(Deserialize, JsonSchema)]
+struct InstanceLookupQueryParam {
+    organization_name: Option<Name>,
+    project_name: Option<Name>,
+}
+
+#[endpoint {
+    method = GET,
+    path = "/instances/{instance}",
+    tags = ["instances"],
+}]
+async fn instance_lookup(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    query_params: Query<InstanceLookupQueryParam>,
+    path_params: Path<InstanceLookupPathParam>,
+) -> Result<HttpResponseOk<Instance>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let query = query_params.into_inner();
+    let instance_ident = &path.instance;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let instance = match instance_ident {
+            ResourceIdentifier::Name(name) => {
+                let organization_name =
+                    query.organization_name.ok_or_else(|| {
+                        Error::InvalidRequest { message: "organization_name is required when using instance name".to_string() }
+                    })?;
+                let project_name = query.project_name.ok_or_else(|| {
+                    Error::InvalidRequest {
+                        message:
+                            "project_name is required when using instance name"
+                                .to_string(),
+                    }
+                })?;
```

Comment:
@ahl would appreciate your input on if this is the right way to validate these params. _Technially_ this implementation will ignore it if you provide params for project/org and a UUID.